src/read.c

```diff
@@ -3571,33 +3571,39 @@ static avifResult avifParseMinimizedImageBox(avifMeta * meta, uint64_t rawOffset
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasAlpha, 1), AVIF_RESULT_BMFF_PARSE_FAILED);              // bit(1) alpha_flag;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasExplicitCicp, 1), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(1) explicit_cicp_flag;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasHdr, 1), AVIF_RESULT_BMFF_PARSE_FAILED);          // bit(1) hdr_flag;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasIcc, 1), AVIF_RESULT_BMFF_PARSE_FAILED);          // bit(1) icc_flag;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasExif, 1), AVIF_RESULT_BMFF_PARSE_FAILED);         // bit(1) exif_flag;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasXmp, 1), AVIF_RESULT_BMFF_PARSE_FAILED);          // bit(1) xmp_flag;
-
-    uint32_t chromaSubsampling, orientation;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &chromaSubsampling, 2), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(2) chroma_subsampling;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &orientation, 3), AVIF_RESULT_BMFF_PARSE_FAILED);       // bit(3) orientation_minus1;
+    const avifBool hasExplicitCodecTypes = (data[0] & 0x20) != 0; // bit(1) explicit_codec_types_flag;
+    const avifBool floatFlag = (data[0] & 0x10) != 0;             // bit(1) float_flag;
+    const avifBool fullRange = (data[0] & 0x8) != 0;              // bit(1) full_range_flag;
+    const avifBool hasAlpha = (data[0] & 0x4) != 0;               // bit(1) alpha_flag;
+    const avifBool hasExplicitCicp = (data[0] & 0x2) != 0;        // bit(1) explicit_cicp_flag;
+    const avifBool hasHdr = (data[0] & 0x1) != 0;                 // bit(1) hdr_flag;
+    const avifBool hasIcc = (data[1] & 0x80) != 0;                // bit(1) icc_flag;
+    const avifBool hasExif = (data[1] & 0x40) != 0;               // bit(1) exif_flag;
+    const avifBool hasXmp = (data[1] & 0x20) != 0;                // bit(1) xmp_flag;
+
+    const uint32_t chromaSubsampling = (data[1] & 0x18) >> 3; // bit(2) chroma_subsampling;
+    uint32_t orientation = data[1] & 0x7;                     // bit(3) orientation_minus1;
     ++orientation;
 
     // Spatial extents
-    uint32_t smallDimensionsFlag, width, height;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &smallDimensionsFlag, 1), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(1) small_dimensions_flag;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &width, smallDimensionsFlag ? 7 : 15),
-                  AVIF_RESULT_BMFF_PARSE_FAILED); // unsigned int(small_dimensions_flag ? 7 : 15) width_minus1;
+    const avifBool smallDimensionsFlag = (data[2] & 0x80) != 0; // bit(1) small_dimensions_flag;
+    // unsigned int(small_dimensions_flag ? 7 : 15) width_minus1;
+    uint32_t width = data[2];
+    if (smallDimensionsFlag) {
+        width &= 0x7f;
+    } else {
+        AVIF_CHECKERR(avifROStreamRead(&s, data, 1), AVIF_RESULT_BMFF_PARSE_FAILED);
+        width = (width << 8) + data[0];
+    }
     ++width;
+    uint32_t height;
```

Comment:
The easy parsing ends with the `width_minus1` field. There are two ways to extend the easy parsing to the `height_minus1` field:
1. Waste one bit and split `small_dimensions_flag` into `small_width_flag` and `small_height_flag`, or
2. Move the `few_codec_config_bytes_flag` field before the `height_minus1` field.

Reply:
@podborski Dimitri: Please see this comment. I just thought of a third way to extend byte alignment to the `height_minus1` field:
* Change `small_dimensions_flag ? 7 : 15` to `small_dimensions_flag ? 8 : 16` for `width_minus1` and `height_minus1`. One justification for this is that the maximum AV1 frame width and height are 2^16. This will require moving one of the boolean flags in the first two bytes later, so that `small_dimensions_flag` can be moved to the end of the second byte.

Reply:
Thanks for catching this @wantehchang the intention was indeed to have it byte aligned including the dimension signaling. We should definitely bring an input contribution on this issue to the next meeting with the possible solutions and our preference for one of the solutions. To me it feels like having a separate flag for width and height is the better choice as there could be cases where only one of them can use 7 bits instead of 15. I'm not sure if extending the range for dimensions could be well accepted as the main use-case for slimHEIF was to target small files. Also slimHEIF is codec agnostic, not sure if we can motivate using AV1 limitations.

Reply:
@podborski Thanks for the confirmation. Note that `small_dimensions_flag` is used to control the size of the `gainmap_width_minus1` and `gainmap_height_minus1` fields later in the mini box., but byte alignment is already lost at that point. So I don't think we need to worry about the byte alignment of those two fields.